<a href="https://colab.research.google.com/github/ABDELLAH-Hallou/BigMart-outlet-sales-prediction/blob/master/BigMart_outlet_sales_prediction.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# **Project Description** : BigMart outlets Sales Prediction







#### The data scientists at BigMart have collected 2013 sales data for 1559 products across 10 stores in different cities. Also, certain attributes of each product and store have been defined. The aim of this data science project is to build a predictive model and find out the sales of each product at a particular store.
## Our objective
#### Using this model, BigMart will try to understand the properties of products and stores which play a key role in increasing sales.

##We will handle this problem in a structured way. We will be following the table of content given below:

### 1.   Hypothesis Generation
### 2.   Loading Packages and Data
###3.   Data Structure and Content
###4.   Exploratory Data Analysis
###5.   Univariate Analysis
###6.   Bivariate Analysis
###7.   Missing Value Treatment
###8.   Feature Engineering
###9.   Encoding Categorical Variables
###10.   Label Encoding
###11.   One Hot Encoding
###12.   PreProcessing Data
###13.   Modeling
###14.   Summary

#**Hypothesis Generation**


##Features based on the Store : 
####1.   The city in which the store is located : 
*   Stores located in urban cities should have higher sales compared to stores in rural areas
*   Stores located in big cities should have higher sales compared to stores located in small cities.

####2.   The location of the store in the city :
*   Stores located in the city center should have higher sales compared to stores on the outskirts of the city.

####3.   Competitor stores :
*   Stores close to competitor stores should sell less than other stores far away from competitor stores.

####4.   Size of the store :
*   Large stores should have higher sales than medium and small stores.

####5.   Store design and architecture: 
*   Well-designed stores can attract customers.

####6.   Marketing: 
*   Stores having a good marketing division can attract customers through the right offers.

##Features based on the Product : 
####1.   Product Utility:
*   Daily use products have a higher tendency to sell compared to other products.

####2.   Product Quality
*   The quality of the product and its packaging can attract customers and sell more.

####3.   Product Visibility in the store: 
*   Products that are placed in an attention-catching place should have higher sales.

####4.   Product Branding: 
*   Branded products have more trust of the customers so they should have high sales.

##Features based on the Customer : 
####1.   Job profile and annual income: 
*   A customer with a stable job and high income should have higher purchases.

####2.   Family size: 
*   A customer with a large family should have higher purchases

# **Loading Packages and Data**

In [9]:
# Loading Packages and Data
import csv
import numpy as np
import pandas as pd
filename = '/content/sample_data/Bigmart_Train.csv'
BigMartTrainData = pd.read_csv(filename, index_col='Item_Identifier')
BigMartTrainData

Item_Weight Item_Fat_Content  Item_Visibility  \
Item_Identifier                                                  
FDA15                  9.300          Low Fat         0.016047   
DRC01                  5.920          Regular         0.019278   
FDN15                 17.500          Low Fat         0.016760   
FDX07                 19.200          Regular         0.000000   
NCD19                  8.930          Low Fat         0.000000   
...                      ...              ...              ...   
FDF22                  6.865          Low Fat         0.056783   
FDS36                  8.380          Regular         0.046982   
NCJ29                 10.600          Low Fat         0.035186   
FDN46                  7.210          Regular         0.145221   
DRG01                 14.800          Low Fat         0.044878   

                             Item_Type  Item_MRP Outlet_Identifier  \
Item_Identifier                                                      
FDA15                            Dairy  249.8092            OUT049   
DRC01                      Soft Drinks   48.2692            OUT018   
FDN15                             Meat  141.6180            OUT049   
FDX07            Fruits and Vegetables  182.0950            OUT010   
NCD19                        Household   53.8614            OUT013   
...                                ...       ...               ...   
FDF22                      Snack Foods  214.5218            OUT013   
FDS36                     Baking Goods  108.1570            OUT045   
NCJ29               Health and Hygiene   85.1224            OUT035   
FDN46                      Snack Foods  103.1332            OUT018   
DRG01                      Soft Drinks   75.4670            OUT046   

                 Outlet_Establishment_Year Outlet_Size Outlet_Location_Type  \
Item_Identifier                                                               
FDA15                                 1999      Medium               Tier 1   
DRC01                                 2009      Medium               Tier 3   
FDN15                                 1999      Medium               Tier 1   
FDX07                                 1998         NaN               Tier 3   
NCD19                                 1987        High               Tier 3   
...                                    ...         ...                  ...   
FDF22                                 1987        High               Tier 3   
FDS36                                 2002         NaN               Tier 2   
NCJ29                                 2004       Small               Tier 2   
FDN46                                 2009      Medium               Tier 3   
DRG01                                 1997       Small               Tier 1   

                       Outlet_Type  Item_Outlet_Sales  
Item_Identifier                                        
FDA15            Supermarket Type1          3735.1380  
DRC01            Supermarket Type2           443.4228  
FDN15            Supermarket Type1          2097.2700  
FDX07                Grocery Store           732.3800  
NCD19            Supermarket Type1           994.7052  
...                            ...                ...  
FDF22            Supermarket Type1          2778.3834  
FDS36            Supermarket Type1           549.2850  
NCJ29            Supermarket Type1          1193.1136  
FDN46            Supermarket Type2          1845.5976  
DRG01            Supermarket Type1           765.6700  

[8523 rows x 11 columns]

## Data Dictionary
#### We have train 8523 dataset, which contains both input and output variables,and test 5681 dataset, We need to predict the sales for these.
*   **Item_Identifier**: Product ID
*   **Item_Weight**: Product Weight 
*   **Item_Fat_Content**: Is the product Low Fat or not(Regular)
*   **Item_Visibility**: The percentage of total display area for all products in the store allocated to each product
*   **Item_Type**: Product Category
*   **Item_MRP**: Maximum Retail Price of the Product
*   **Outlet_Identifier**: Store ID
*   **Outlet_Establishment_Year**: The year in which the Store was established
*   **Outlet_Size**: The area of ground space covered by the store (Small-Medium-High)
*   **Outlet_Location_Type**: The type of city in which the store is located
*   **Outlet_Type**: The outlet Category (Grocery store or some sort of supermarket)
*   **Item_Outlet_Sales**: Sales of the product in the particulat store. This is the outcome variable to be predicted.